packages/compartment-mapper/src/import-archive.js

```diff
@@ -0,0 +1,80 @@
+// @ts-check
+
+import { defaultParserForLanguage } from './import-archive-parsers.js';
+import {
+  parseArchive as parseArchiveLite,
+  loadArchive as loadArchiveLite,
+  importArchive as importArchiveLite,
+} from './import-archive-lite.js';
+
+const { assign, create, freeze } = Object;
+
+/** @import {Application, ComputeSourceLocationHook, ComputeSourceMapLocationHook, ExecuteOptions, ExitModuleImportHook, HashFn, LoadArchiveOptions, ReadPowers} from './types.js' */
+/** @import {ParserForLanguage} from './types.js' */
+
+// Have to give it a name to capture the external meaning of Compartment
+// Otherwise @param {typeof Compartment} takes the Compartment to mean
+// the const variable defined within the function.
+//
```

Comment:
(suggestion) IMO it isn't necessary to explain this, but nbd.  I would probably even just zap the `@typedef` altogether and use `typeof Compartment` in the `parseArchive` signature's docstring

Reply:
Using `typedef Compartment` doesn’t work for some reason, thus the comment. My preference as well.

Reply:
@kriskowal FWIW I cannot reproduce whatever the problem was that you ran into.  This compiles cleanly for me:

```js
/**
 * @typedef {object} Options
 * @property {string} [expectedSha512]
 * @property {HashFn} [computeSha512]
 * @property {Record<string, unknown>} [modules]
 * @property {ExitModuleImportHook} [importHook]
 * @property {typeof Compartment} [Compartment]
 * @property {ComputeSourceLocationHook} [computeSourceLocation]
 * @property {ComputeSourceMapLocationHook} [computeSourceMapLocation]
 * @property {ParserForLanguage} [parserForLanguage]
 */
```

Reply:
Yeah, there are `Compartment` compatible constructors created by wrapping the `Compartment` constructor. Notably, `Compartment` before lockdown is different than after. Every `Compartment` has its own instance of `Compartment`. They’re structurally compatible, though. Unclear to me why that would make a difference to types, but they do have different values.

Reply:
even though TS is structural, because `Compartment` is a function, it will only ever allow the global `Compartment` from whichever version of `ses` provides it.  That may be fine, but also seems redundant.  Though it may be useful if we're using some mocked-out `Compartment` in tests?

Reply:
```
src/import-archive-lite.js:266:4 - error TS2502: 'Compartment' is referenced directly or indirectly in its own type annotation.

266  * @param {typeof Compartment} [options.Compartment]
```

Reply:
ah, I see that now.  It's because of the `DefaultCompartment`.  `typeof DefaultCompartment` works